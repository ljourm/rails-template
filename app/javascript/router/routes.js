import NotFound from "pages/NotFound"

const Foo = { template: "<div>foo</div>" }
const Bar = { template: "<div>bar</div>" }

export default [
  { path: "/foo/:id", component: Foo },
  { path: "/bar", component: Bar },
  { path: "*", component: NotFound },
]
