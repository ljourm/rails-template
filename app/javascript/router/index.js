import Vue from "vue/dist/vue.esm.js"
import VueRouter from "vue-router"
Vue.use(VueRouter)

import routes from "./routes"

export default new VueRouter({
  routes,
})
