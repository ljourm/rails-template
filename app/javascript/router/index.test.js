import router from "."

describe("router", () => {
  test("router", () => {
    expect(router).toBeTruthy()
  })

  test("snapshot", () => {
    // NOTE: スナップショットの更新が頻繁に発生するためコメントアウト
    // expect(router).toMatchSnapshot()
  })
})
