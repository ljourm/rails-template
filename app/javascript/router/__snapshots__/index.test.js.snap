// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`router snapshot 1`] = `
VueRouter {
  "afterHooks": Array [],
  "app": null,
  "apps": Array [],
  "beforeHooks": Array [],
  "fallback": false,
  "history": HTML5History {
    "base": "",
    "current": Object {
      "fullPath": "/",
      "hash": "",
      "matched": Array [],
      "meta": Object {},
      "name": null,
      "params": Object {},
      "path": "/",
      "query": Object {},
    },
    "errorCbs": Array [],
    "pending": null,
    "ready": false,
    "readyCbs": Array [],
    "readyErrorCbs": Array [],
    "router": [Circular],
  },
  "matcher": Object {
    "addRoutes": [Function],
    "match": [Function],
  },
  "mode": "history",
  "options": Object {
    "mode": "history",
    "routes": Array [
      Object {
        "component": Object {
          "beforeRouteEnter": [Function],
          "beforeRouteUpdate": [Function],
          "components": Object {
            "UserForm": Object {
              "data": [Function],
              "methods": Object {
                "destroy": [Function],
                "submit": [Function],
              },
              "props": Object {
                "user": Object {
                  "required": true,
                  "type": [Function],
                },
              },
              "render": [Function],
              "staticRenderFns": Array [
                [Function],
              ],
              "watch": Object {
                "user": Object {
                  "handler": [Function],
                  "immediate": true,
                },
              },
            },
          },
          "computed": Object {
            "user": [Function],
          },
          "methods": Object {
            "destroy": [Function],
            "destroyData": [Function],
            "fetch": [Function],
            "submit": [Function],
            "submitForm": [Function],
          },
          "render": [Function],
          "staticRenderFns": Array [],
        },
        "path": "/users/:id",
      },
      Object {
        "component": Object {
          "beforeRouteEnter": [Function],
          "components": Object {
            "UserList": Object {
              "components": Object {
                "EmailLink": Object {
                  "computed": Object {
                    "href": [Function],
                  },
                  "props": Object {
                    "email": Object {
                      "required": true,
                      "type": [Function],
                    },
                  },
                  "render": [Function],
                  "staticRenderFns": Array [],
                },
              },
              "computed": Object {
                "users": [Function],
              },
              "render": [Function],
              "staticRenderFns": Array [],
            },
          },
          "computed": Object {
            "users": [Function],
          },
          "methods": Object {
            "fetch": [Function],
          },
          "render": [Function],
          "staticRenderFns": Array [],
        },
        "path": "/users",
      },
      Object {
        "component": Object {
          "template": "<div>foo</div>",
        },
        "path": "/foo/:id",
      },
      Object {
        "component": Object {
          "template": "<div>bar</div>",
        },
        "path": "/bar",
      },
      Object {
        "component": Object {
          "render": [Function],
          "staticRenderFns": Array [],
        },
        "path": "*",
      },
    ],
  },
  "resolveHooks": Array [],
}
`;
